packages/ui-source-code-editor/src/SourceCodeEditor/SearchPanel.tsx

```diff
@@ -0,0 +1,136 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015 - present Instructure, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import React, { useState } from 'react'
+import { createRoot } from 'react-dom/client'
+import {
+  setSearchQuery,
+  search,
+  findNext,
+  findPrevious,
+  SearchQuery,
+  closeSearchPanel
+} from '@codemirror/search'
+import { EditorView } from '@codemirror/view'
+import { TextInput } from '@instructure/ui-text-input'
+import { IconButton } from '@instructure/ui-buttons'
+import {
+  IconArrowOpenDownLine,
+  IconArrowOpenUpLine,
+  IconSearchLine
+} from '@instructure/ui-icons'
+
+function SearchPanel({ view }: { view: EditorView }) {
+  const [searchQueryStr, setSearchQueryStr] = useState('')
+
+  const handleChange = (
+    _e: React.ChangeEvent<HTMLInputElement>,
+    value: string
+  ) => {
+    setSearchQueryStr(value)
+    handleHighlightSearch(value)
+  }
+
+  const handleHighlightSearch = (searchStr: string) => {
+    view.dispatch({
+      effects: setSearchQuery.of(
+        new SearchQuery({
+          search: searchStr
+        })
+      )
+    })
+  }
+
+  const handleKeyDown = (e: React.KeyboardEvent) => {
+    if (e.key !== 'Enter') return
+    if (!e.shiftKey) handleFindNext()
+    else handleFindPrev()
+  }
+
+  const handleKeyUp = (e: React.KeyboardEvent) => {
+    if (e.key !== 'Escape') return
+    closeSearchPanel(view)
+  }
+
+  const handleFindNext = () => {
+    handleHighlightSearch(searchQueryStr)
+    findNext(view)
+  }
+
+  const handleFindPrev = () => {
+    handleHighlightSearch(searchQueryStr)
+    findPrevious(view)
+  }
+
+  return (
+    <div className="search-panel">
+      <TextInput
+        inputRef={(r) => r?.focus()}
+        size="small"
+        display="inline-block"
+        width="20rem"
+        placeholder="Find in code..."
```

Comment:
We never hardcode text in our components, they are almost always used by a i18n library. Please expose this to a required prop.

Reply:
in that case should the search panel be disabled by default to limit the number of required props? and instead of a boolean, the prop could be the search panel config

Reply:
hmm good questions. I would not disable it by default, its better that nothing. I also like the idea of a search panel config object, if its `null`/`undefined` its not rendered, and has these texts that are mandatory